datamap/mv/core/src/main/scala/org/apache/carbondata/mv/datamap/MVHelper.scala

```diff
@@ -152,6 +176,33 @@ object MVHelper {
           fields,
           fieldRelationMap,
           tableProperties)
+      if (timeSeriesFunction != null) {
+        if (null != mainTable && (parentTablesList.size() == 1)) {
+          if (!mainTable.getTableName.equalsIgnoreCase(parentTablesList.get(0).getTableName)) {
+            throw new MalformedCarbonCommandException(
+              "Parent table name is different in Create and Select Statement")
+          }
+        }
+        val timeSeriesDataType = parentTablesList
+          .get(0)
+          .getTableInfo
+          .getFactTable
+          .getListOfColumns
+          .asScala
+          .filter(columnSchema => columnSchema.getColumnName
+            .equalsIgnoreCase(timeSeriesColumn))
+          .head
+          .getDataType
+        if (timeSeriesDataType.equals(DataTypes.DATE) ||
+            timeSeriesDataType.equals(DataTypes.TIMESTAMP)) {
+          if (timeSeriesDataType.equals(DataTypes.DATE)) {
+            TimeSeriesUtil.validateTimeSeriesGranularityForDate(timeSeriesFunction)
+          }
```

Comment:
No granularity validation for timestamp ?

Reply:
Validation is already added for granularity. This validation is particularly added for Date type, as for Date type, granularity like second,hour,minute is invalid.